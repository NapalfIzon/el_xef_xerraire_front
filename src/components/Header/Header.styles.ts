import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    marginBottom: 15,
  },
  logo: {
    width: "25%",
    height: 100,
  },
  title: {
    width: "50%",
  },
});
