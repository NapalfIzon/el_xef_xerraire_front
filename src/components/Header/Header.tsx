import React from "react";
import { Center, Image, HStack, View } from "native-base";
import styles from "./Header.styles";

const Header = () => {
  return (
    <View>
      <Center>
        <HStack style={styles.container}>
          <Image
            style={styles.logo}
            source={{
              uri: "https://i.ibb.co/SKV5Btw/logo-xef-001.webp",
            }}
            alt="Xef Xerraire Logo"
            size="xs"
          />
          <Image
            style={styles.title}
            source={{
              uri: "https://i.ibb.co/q5MTyzd/title-004.png",
            }}
            alt="El Xef Xerraire"
            size="xs"
          />
          <Image
            style={styles.logo}
            source={{
              uri: "https://i.ibb.co/SKV5Btw/logo-xef-001.webp",
            }}
            alt="Xef Xerraire Logo"
            size="xs"
          />
        </HStack>
      </Center>
    </View>
  );
};

export default Header;
