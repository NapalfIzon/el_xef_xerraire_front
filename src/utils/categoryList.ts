export const categoryListObject = {
  Aperitivos: "Aperitivos",
  Tapas: "Tapas",
  Arroces: "Arroces",
  Cereales: "Cereales",
  Aves: "Aves",
  Carne: "Carne",
  Cócteles: "Cócteles",
  Bebidas: "Bebidas",
  Ensaladas: "Ensaladas",
  Guisos: "Guisos",
  Huevos: "Huevos",
  Lácteos: "Lácteos",
  Legumbres: "Legumbres",
  Mariscos: "Mariscos",
  Pan: "Pan",
  Bollería: "Bollería",
  Pasta: "Pasta",
  Pescado: "Pescado",
  Postres: "Postres",
  Salsas: "Salsas",
  Sopas: "Sopas",
  Cremas: "Cremas",
  Verduras: "Verduras",
};

export const categoryList = [
  "Aperitivos",
  "Tapas",
  "Arroces",
  "Cereales",
  "Aves",
  "Carne",
  "Cócteles",
  "Bebidas",
  "Ensaladas",
  "Guisos",
  "Huevos",
  "Lácteos",
  "Legumbres",
  "Mariscos",
  "Pan",
  "Bollería",
  "Pasta",
  "Pescado",
  "Postres",
  "Salsas",
  "Sopas",
  "Cremas",
  "Verduras",
];
