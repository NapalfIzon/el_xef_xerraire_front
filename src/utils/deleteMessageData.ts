import { IAlertMessage } from "../interfaces/componentsInterface";

export const deleteRecipeMessageText: IAlertMessage = {
  title: "Borrar receta?",
  body: "Esta acci칩n borrar치 tu receta, y no podr치 ser recuperada.",
  close: "Vuelve atr치s",
  action: "Borrar",
};
