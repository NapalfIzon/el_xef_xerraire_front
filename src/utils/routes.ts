export const backgroundImage = "../../../images/logo_xef_001.webp";
