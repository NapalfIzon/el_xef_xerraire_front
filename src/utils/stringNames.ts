export const stringName = {
  keyName: "token",
};
