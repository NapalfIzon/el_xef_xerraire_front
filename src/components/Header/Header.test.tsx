import React from "react";
import { render } from "@testing-library/react-native";
import Header from "./Header";
import { NativeBaseProvider } from "native-base";
import { inset } from "../../utils/nativeBaseInset";

describe("Given a Header component,", () => {
  describe("When it is rendered,", () => {
    test("Then it should render a logo image and a title image.", () => {
      const result = render(
        <NativeBaseProvider initialWindowMetrics={inset}>
          <Header />
        </NativeBaseProvider>
      );

      expect(result).toMatchSnapshot();
    });
  });
});
