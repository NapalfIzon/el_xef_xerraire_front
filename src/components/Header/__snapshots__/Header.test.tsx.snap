// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Header component, When it is rendered, Then it should render a logo image and a title image. 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "marginBottom": 15,
            },
          ]
        }
      >
        <Image
          accessibilityLabel="Xef Xerraire Logo"
          alt="Xef Xerraire Logo"
          onError={[Function]}
          source={
            Object {
              "uri": "https://i.ibb.co/SKV5Btw/logo-xef-001.webp",
            }
          }
          style={
            Array [
              Object {
                "height": 40,
                "maxWidth": "100%",
                "width": 40,
              },
              Object {
                "height": 100,
                "width": "25%",
              },
            ]
          }
        />
        <View
          style={Object {}}
        />
        <Image
          accessibilityLabel="El Xef Xerraire"
          alt="El Xef Xerraire"
          onError={[Function]}
          source={
            Object {
              "uri": "https://i.ibb.co/q5MTyzd/title-004.png",
            }
          }
          style={
            Array [
              Object {
                "height": 40,
                "maxWidth": "100%",
                "width": 40,
              },
              Object {
                "width": "50%",
              },
            ]
          }
        />
        <View
          style={Object {}}
        />
        <Image
          accessibilityLabel="Xef Xerraire Logo"
          alt="Xef Xerraire Logo"
          onError={[Function]}
          source={
            Object {
              "uri": "https://i.ibb.co/SKV5Btw/logo-xef-001.webp",
            }
          }
          style={
            Array [
              Object {
                "height": 40,
                "maxWidth": "100%",
                "width": 40,
              },
              Object {
                "height": 100,
                "width": "25%",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
